src/transformers/trainer.py

```diff
@@ -493,19 +501,30 @@ def __init__(
                     "Please do not set arguments related to `mixed_precision`"
                 )
             if args.half_precision_backend == "auto":
-                if _is_native_amp_available:
-                    args.half_precision_backend = "amp"
-                else:
-                    if args.bf16:
-                        raise ValueError("Tried to use `bf16` but native amp is not available")
+                if args.device == torch.device("cuda"):
```

Comment:
This test is completely unreliable and will be false if `args.device` is set with `torch.device(0)` or when there are multiple GPUs. It should therefore be removed.

Reply:
Got it, have made changes to check if it is CPU or others ( GPU, multiple GPU...), thanks.